### PR TITLE
Feature/dashboard resturant filter

### PR DESCRIPTION
**Pull request by mujtaba912**

### CODE REVIEW

**Review by HaidyAbuGom3a — APPROVED** [review]

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            selectedIcon = painterResource(if (isSystemInDarkTheme()) "ic_filled_star_dark.svg" else "ic_filled_star_light.svg"),
+            halfSelectedIcon = painterResource(if (isSystemInDarkTheme()) "ic_half_filled_star_dark.svg" else "ic_half_filled_star_light.svg"),
+            notSelectedIcon = painterResource(if (isSystemInDarkTheme()) "ic_star_dark.svg" else "ic_star_light.svg"),
+            iconsSize = Theme.dimens.space24,
+            iconsPadding = PaddingValues(horizontal = Theme.dimens.space8),
+            modifier = Modifier.fillMaxWidth().padding(top = Theme.dimens.space16)
+                .background(color = Theme.colors.background)
+                .padding(horizontal = Theme.dimens.space24, vertical = Theme.dimens.space16),
+            onClick = { onClickRating(it) }
+        )
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt:85]: refactor code, don't make it reach the line

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt:1]: dropdown is one word thus DropdownMenu

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            count = 3,
+            icon = painterResource("ic_dollar_sign.svg"),
+            enabledIconsColor = Theme.colors.success,
+            disabledIconsColor = Theme.colors.disable,
+            iconsPadding = PaddingValues(horizontal = Theme.dimens.space8),
+            iconsSize = Theme.dimens.space16,
+            modifier = Modifier.fillMaxWidth().padding(top = Theme.dimens.space16)
+                .background(color = Theme.colors.background)
+                .padding(horizontal = Theme.dimens.space24, vertical = Theme.dimens.space16),
+            onClick = { onClickPrice(it) }
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt:104]:
```suggestion
            onClick = ::onClickPrice
```
pass as a refrance

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import org.thechance.common.presentation.composables.EditableRatingBar
+
+@Composable
+fun RestaurantFilterDropDownMenu(
+    expanded: Boolean,
+    rating: Double,
+    priceLevel: Int,
+    onClickRating: (Double) -> Unit,
+    onClickPrice: (Int) -> Unit,
+    onDismissRequest: () -> Unit
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt:31]: rearrange so functions are at the beginning

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt (hunk 2 above)
Comment (mujtaba912) [reply client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt:104]: its not supported

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt (hunk 3 above)
Comment (mujtaba912) [reply client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt]: done

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt
Comment (mujtaba912) [reply client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantFilterDropDownMenu.kt:1]: done

**Review by Saif-Alhaider — APPROVED** [review]

**Pull request merged**